scene/3d/gpu_particles_3d.cpp

```diff
@@ -149,6 +149,7 @@ void GPUParticles3D::set_process_material(const Ref<Material> &p_material) {
 	RID material_rid;
 	if (process_material.is_valid()) {
 		material_rid = process_material->get_rid();
+	} else if (cast_to<ParticleProcessMaterial>(*process_material)) {
 		process_material->connect("emission_shape_changed", callable_mp((Node3D *)this, &GPUParticles3D::update_gizmos));
 	}
```

Comment:
This won't work, since the `process_material.is_valid()` condition is true, so it will never evaluate the `else if`.
It should be:
```suggestion
	if (process_material.is_valid()) {
		material_rid = process_material->get_rid();
		if (Ref<ParticleProcessMaterial>(process_material).is_valid()) {
			process_material->connect("emission_shape_changed", callable_mp((Node3D *)this, &GPUParticles3D::update_gizmos));
		}
	}
```